src/test/scala/uk/gov/nationalarchives/Steps.scala

```diff
@@ -0,0 +1,182 @@
+    val metadata = TREMetadata(TREMetadataParameters(parser, tre, tdr))
+    metadata.asJson.printWith(Printer.noSpaces).getBytes
+  }
+
+  private def parentPath(entityRef: Option[UUID], paths: List[String] = Nil): IO[List[String]] = {
+    if (entityRef.isEmpty) {
+      IO(paths)
+    } else {
+      for {
+        entityClient <- entityClient(url, secretName)
+        entity <- entityClient.getEntity(entityRef.get, StructuralObject)
+        res <- parentPath(entity.parent, entity.title.getOrElse("") :: paths)
+      } yield res
+    }
+  }
+
+  def getParentPath(entityRef: Option[UUID]): String = {
+    parentPath(entityRef, Nil).unsafeRunSync().mkString("/")
+  }
+
+  def getIdentifiers(potentialRef: Option[UUID]): Map[String, String] = (for {
+    ref <- IO.fromOption(potentialRef)(new Exception("No parent ref provided"))
+    entityClient <- entityClient(url, secretName)
+    entity <- entityClient.getEntity(ref, StructuralObject)
+    identifiers <- entityClient.getEntityIdentifiers(entity)
+  } yield identifiers.map(id => id.identifierName -> id.value).toMap).unsafeRunSync()
+
+  def getEntity(title: String): Entity = (for {
+    contentClient <- contentClient(url, secretName)
+    entityClient <- entityClient(url, secretName)
+    res <- contentClient.searchEntities(SearchQuery("", SearchField("xip.title", List(title)) :: Nil))
+    entities <- res.map(entity => entityClient.getEntity(entity.ref, entity.entityType.get)).sequence
+    entity <- IO.fromOption(entities.headOption)(new Exception("Entity not found"))
+    _ <- Logger[IO].info(s"Entity ${entity.ref} found for title $title")
+  } yield entity).unsafeRunSync
+
+  def sendSQSMessage(reference: UUID, skipSeriesLookup: Boolean = false): SendMessageResponse = {
+    val parameter: Parameter = if (skipSeriesLookup) {
+      SQSParameterWithSkipLookup("OK", reference.toString, bucket, s"$reference.tar.gz", skipSeriesLookup = true)
+    } else {
+      SQSParameter("OK", reference.toString, bucket, s"$reference.tar.gz")
+    }
+    val sqsMessage = SQSMessage(parameter)
+    daSQSClient.sendMessage(sqsUrl)(sqsMessage)
+  }.unsafeRunSync
+
+  def createJudgment(cite: Option[String], uri: Option[String]): UUID = {
+    val reference = UUID.randomUUID()
+    val docxBytes = createDocx(reference)
+    val checksum = createChecksum(docxBytes)
+    val metadataBytes = createMetadata(cite, uri, reference, checksum)
+    val tarBytes = tarFile(reference, docxBytes, metadataBytes)
+    for {
+      _ <- Logger[IO].info(s"Creating judgment with reference $reference")
+      reference <- Stream.emits[IO, Byte](tarBytes).chunks.map(_.toByteBuffer).toUnicastPublisher.use { publisher =>
+        daS3Client.upload(bucket, s"$reference.tar.gz", tarBytes.length, publisher).map(_ => reference)
+      }
+    } yield reference
+  }.unsafeRunSync()
+
+  def tarFile(reference: UUID, docxBytes: Array[Byte], metadataBytes: Array[Byte]): Array[Byte] = {
```

Comment:
Maybe `createTarFiles` or `tarFiles`?

Reply:
I've gone with `createTarFile` as we're only creating one at a time.